### PR TITLE
Added gradio example to docs

### PR DESCRIPTION
**Pull request by merveenoyan**
cc @osanseviero

### CODE REVIEW

**Comment by merveenoyan** [comment]
Also ping @yvrjsharma

File: docs/source/basic_tutorials/consuming_tgi.md [hunk 1]
```diff
@@ -75,6 +75,44 @@ To serve both ChatUI and TGI in same environment, simply add your own endpoints
 
 ![ChatUI](https://huggingface.co/datasets/huggingface/documentation-images/resolve/main/chatui_screen.png)
 
+## Gradio
+
+Gradio has a `ChatInterface` class to create neat UIs for chatbots. Let's take a look at how to create a chatbot with streaming mode using TGI and Gradio. Assume you are serving your model on port 8080.
```
Comment (osanseviero) [inline docs/source/basic_tutorials/consuming_tgi.md:80]: Same as you explained what ChatUI is, I would have an intro sentence explaining what Gradio is and linking to its docs.

File: docs/source/basic_tutorials/consuming_tgi.md [hunk 2]
```diff
@@ -75,6 +75,44 @@ To serve both ChatUI and TGI in same environment, simply add your own endpoints
...
+
+```python
+import gradio as gr
+from huggingface_hub import InferenceClient
```
Comment (osanseviero) [inline docs/source/basic_tutorials/consuming_tgi.md:84]: Should we add a pip install for these two before?

File: docs/source/basic_tutorials/consuming_tgi.md (hunk 1 above)
Comment (osanseviero) [inline docs/source/basic_tutorials/consuming_tgi.md:80]: Given you use `huggingface_hub`, I would add a link to the `consuming_tgi` section on `huggingface_hub` client

File: docs/source/basic_tutorials/consuming_tgi.md [hunk 3]
```diff
@@ -75,6 +75,44 @@ To serve both ChatUI and TGI in same environment, simply add your own endpoints
...
+
+# initialize InferenceClient
```
Comment (osanseviero) [inline docs/source/basic_tutorials/consuming_tgi.md:86]: No need for this docstring I think, your code is self-explanatory

File: docs/source/basic_tutorials/consuming_tgi.md [hunk 4]
```diff
@@ -75,6 +75,44 @@ To serve both ChatUI and TGI in same environment, simply add your own endpoints
...
+    examples=["Are tomatoes vegetables?"],
+    retry_btn=None,
+    undo_btn="Undo",
+    clear_btn="Clear",
+).queue().launch()
+```
+
+The UI looks like this 👇 
+
+![Gradio TGI](https://huggingface.co/datasets/huggingface/documentation-images/resolve/main/gradio-tgi.png)
```
Comment (osanseviero) [inline docs/source/basic_tutorials/consuming_tgi.md:111]: Let's put images in https://huggingface.co/datasets/huggingface/documentation-images/tree/main/tgi please :pray: will be easier to maintain

Comment (osanseviero) [inline docs/source/basic_tutorials/consuming_tgi.md:111]: If possible, please have an image in dark and another image in light mode. Then you can show the image according to the user theme. (check the streaming guide to see how)

Comment (osanseviero) [inline docs/source/basic_tutorials/consuming_tgi.md:111]: Maybe you could embed the gradio demo directly here, as done in the streaming guide

**Review by osanseviero — COMMENTED** [review]
Looking very cool!

File: docs/source/basic_tutorials/consuming_tgi.md (hunk 4 above)
Comment (merveenoyan) [reply docs/source/basic_tutorials/consuming_tgi.md:111]: I'd have to keep the inference endpoints up all the time, do you think it's worth it?

**Review by osanseviero — APPROVED** [review]
Looks great!

**Comment by merveenoyan** [comment]
@osanseviero I replaced screenshots with LLaMA 2 chat since the model is more suitable for chat UI. I also added the Space embedding.

File: docs/source/basic_tutorials/consuming_tgi.md [hunk 5]
```diff
@@ -75,6 +75,62 @@ To serve both ChatUI and TGI in same environment, simply add your own endpoints
...
+    />
+    <img 
+        class="hidden dark:block" 
+        src="https://huggingface.co/datasets/huggingface/documentation-images/resolve/main/tgi/gradio-tgi-dark.png"
+    />
+</div>
+
+You can try the demo directly here 👇 
+
+<gradio-app space="merve/gradio-tgi-2"></gradio-app>
```
Comment (osanseviero) [inline docs/source/basic_tutorials/consuming_tgi.md:128]: Please have it in dark/light mode as https://github.com/huggingface/text-generation-inference/blob/main/docs/source/conceptual/streaming.md?plain=1#L27-L41

File: docs/source/basic_tutorials/consuming_tgi.md [hunk 6]
```diff
@@ -75,6 +75,62 @@ To serve both ChatUI and TGI in same environment, simply add your own endpoints
...
+
+
+You can disable streaming mode using `return` instead of `yield` in your inference function.
```
Comment (osanseviero) [inline docs/source/basic_tutorials/consuming_tgi.md:131]: I think this part is a bit confusing. It needs to change from

```py
for token in client.text_generation(message, max_new_tokens=20, stream=True):
        partial_message += token
        yield partial_message
```

to
```
return client.text_generation(message, max_new_tokens=20):
```

File: docs/source/basic_tutorials/consuming_tgi.md [hunk 7]
```diff
@@ -75,6 +75,81 @@ To serve both ChatUI and TGI in same environment, simply add your own endpoints
 
 ![ChatUI](https://huggingface.co/datasets/huggingface/documentation-images/resolve/main/chatui_screen.png)
 
+## Gradio
+
+Gradio is a Python library that helps you build a web application for your machine learning model with few lines of code. It has a `ChatInterface` class to create neat UIs for chatbots. Let's take a look at how to create a chatbot with streaming mode using TGI and Gradio. Let's install Gradio and Hub Python library first.
```
Comment (osanseviero) [inline docs/source/basic_tutorials/consuming_tgi.md:80]:
```suggestion
Gradio is a Python library that helps you build web applications for your machine learning models with a few lines of code. It has a `ChatInterface` wrapper that helps create neat UIs for chatbots. Let's take a look at how to create a chatbot with streaming mode using TGI and Gradio. Let's install Gradio and Hub Python library first.
```

File: docs/source/basic_tutorials/consuming_tgi.md [hunk 8]
```diff
@@ -75,6 +75,81 @@ To serve both ChatUI and TGI in same environment, simply add your own endpoints
...
+
+```bash
+pip install huggingface-hub gradio
+```
+
+Assume you are serving your model on port 8080, we will query through [InferenceClient](consuming_tgi#inference-Client).
```
Comment (osanseviero) [inline docs/source/basic_tutorials/consuming_tgi.md:86]:
```suggestion
Assume you are serving your model on port 8080, we will query through [InferenceClient](consuming_tgi#inference-client). 
```

File: docs/source/basic_tutorials/consuming_tgi.md [hunk 9]
```diff
@@ -75,6 +75,81 @@ To serve both ChatUI and TGI in same environment, simply add your own endpoints
...
+
+client = InferenceClient(model="http://127.0.0.1:8080")
+
+def inference(message, history):
+    partial_message = ""
+    for token in client.text_generation(message, max_new_tokens=20, stream=True):
+        partial_message += token
+        yield partial_message
+
+gr.ChatInterface(
```
Comment (osanseviero) [inline docs/source/basic_tutorials/consuming_tgi.md:100]: This code does not match the screenshot and the space. The main one I see, which I would change in the Space/screenshots, is `retry_btn="Retry"`.

File: docs/source/basic_tutorials/consuming_tgi.md [hunk 10]
```diff
@@ -75,6 +75,81 @@ To serve both ChatUI and TGI in same environment, simply add your own endpoints
...
+    />
+    <img 
+        class="hidden dark:block" 
+        src="https://huggingface.co/datasets/huggingface/documentation-images/resolve/main/tgi/gradio-tgi-dark.png"
+    />
+</div>
+
+You can try the demo directly here 👇 
+
+<div class="block dark:hidden">
```
Comment (osanseviero) [inline docs/source/basic_tutorials/consuming_tgi.md:128]: I would increase the heigth in the demos, see how it looks in https://moon-ci-docs.huggingface.co/docs/text-generation-inference/pr_867/en/basic_tutorials/consuming_tgi#gradio 

It should not require any iframe scrolling

**Review by osanseviero — APPROVED** [review]
Thanks! I left one change regarding iframes heigth, but otherwise looks good!

**Review by osanseviero — APPROVED** [review]
Very nice! :fire:

**Pull request merged**